test/maintenance_task_api_test.dart

```diff
@@ -0,0 +1,318 @@
+//
+// AUTO-GENERATED FILE, DO NOT MODIFY!
+//
+// @dart=2.12
+
+// ignore_for_file: unused_element, unused_import
+// ignore_for_file: always_put_required_named_parameters_first
+// ignore_for_file: lines_longer_than_80_chars
+
+import 'dart:io';
+
+import 'package:collection/collection.dart';
+import 'package:icure_dart_sdk/api.dart';
+import 'package:icure_dart_sdk/crypto/crypto.dart';
+import 'package:test/test.dart';
+import 'package:uuid/uuid.dart';
+import 'package:uuid/uuid_util.dart';
+import 'util/test_utils_backend.dart';
+
+class CodeEquality implements Equality<CodeStubDto> {
+  @override
+  bool equals(CodeStubDto e1, CodeStubDto e2) {
+    return e1.code == e2.code && e1.type == e2.type && e1.version == e2.version;
+  }
+
+  @override
+  int hash(CodeStubDto e) {
+    // TODO: implement hash
+    throw UnimplementedError();
+  }
+
+  @override
+  bool isValidKey(Object? o) {
+    // TODO: implement isValidKey
+    throw UnimplementedError();
+  }
+
+}
+
+/// tests for MaintenanceTaskApi
+void main() {
+  
+  // final TestBackend backend = DockerTestBackend.getInstance(15984, 16044, "icure", "icure", "test", "admin");
+  final TestBackend backend = RemoteTestBackend.getInstance("icuretest", "icuretest", "http://localhost:16043");
+  final Uuid uuid = Uuid();
+  final login = "hcp-${uuid.v4(options: {'rng': UuidUtil.cryptoRNG})}-delegate";
+  HealthcarePartyDto? delegateHcp;
+  MaintenanceTaskApi? maintenanceTaskApi;
+
+  setUpAll(() async {
+    await backend.init();
+
+    final client = ApiClient.basic(backend.iCureURL, backend.iCureUser, backend.iCurePwd);
+    final hcpKeys = generateRandomPrivateAndPublicKeyPair();
+    delegateHcp = await HealthcarePartyApi(client).createHealthcareParty(
+        new HealthcarePartyDto(id: uuid.v4(options: {'rng': UuidUtil.cryptoRNG}), publicKey: hcpKeys.item2, firstName: "test", lastName: "test")
+    );
+    assert(delegateHcp != null);
+    await UserApi(client).createUser(
+        new UserDto(
+            id: "user-${uuid.v4(options: {'rng': UuidUtil.cryptoRNG})}-hcp",
+            login: login,
+            status: UserDtoStatusEnum.ACTIVE,
+            passwordHash: "{R0DLKxxRDxdtpfY542gOUZbvWkfv1KWO9QOi9yvr/2c=}39a484cbf9057072623177422172e8a173bd826d68a2b12fa8e36ff94a44a0d7",
```

Comment:
Create authenticationToken instead of a password with a limited validation time. Also, generate it randomly (uuid for example) instead of reusing the same passwordHash each time